CommonMark formatter with CKEditor5

oliverguenther opened the pull request:
**TODOS**

**Frontend**

- [x] Extract ckeditor build into separate plugin

**Backend**

- [x] Markdown backend
  - [x] Tests
- [X] Separate frontend handling for textile and markdown formatting.

**Out of scope**
- Textile migration
  - Allow plugins to hook into migrator (documents, meetings)
- WP Creation: Image upload tries to immediately upload files, our current pipeline needs to upload attachments after the resource exists.

oliverguenther committed:
CKEditor5 config and bundle for OP

oliverguenther committed:
Introduce commonmark and html-pipeline

oliverguenther committed:
Switchable markdown setting

oliverguenther committed:
Introduce pattern matcher pipeline filter

oliverguenther committed:
Fix legacy formatting module

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/39c7ef215d73b9f99e4e37df9ff2123e96624025) (deployed commit: 101b850943ecce6b5f7ee9bbd86b5da192eb0cc5).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/2d696ff1ce70fa8d4a8ac8ae7b79a8b7dc88c902) (deployed commit: 873928d945611e4e20ee9e24838ad080fe547f4a).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/b5bd1a14f90a37722a72cfa2260c1c5409a1cf49) (deployed commit: 4d3cc59a321d00460f6425172a49f3a85f52dab4).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/2b5e3ef77459d3e7f7235937d37a464e89c35a26) (deployed commit: 4e39fb4a55078400b467b423cd873bf50294719e).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/013503c8343c3b1115ee7e236b1d6375378f4407) (deployed commit: 0c92a7038e5d2b0e73d6451e0ddfb6c3e331a330).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/0093badd5384de2f72fc5031878aa8f784e8a152) (deployed commit: 7fedd55712d634841c5f031b806347cb984594b4).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/1567cc72f83c4168a05cdb97d8cadf714c81ebb3) (deployed commit: 7fedd55712d634841c5f031b806347cb984594b4).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/92a9e5c67cb7e04629268730b7df1c9dbf4e5c1c) (deployed commit: 7fedd55712d634841c5f031b806347cb984594b4).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/71f56ff1294bb86d9d7cc0ca8183e5c2dc75c308) (deployed commit: 7fedd55712d634841c5f031b806347cb984594b4).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/9984f279d262f6510ed2157a9773b442b4cd82ce) (deployed commit: 7fedd55712d634841c5f031b806347cb984594b4).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/63b21a9d36c8bd7c0c2b5e2a6e5dbfd0e7b5a102) (deployed commit: 9fb922c0f518209cedbf4d883c4d9f38a98e604a).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/451df8d82c09da9ca383e7085e082fa3a021114a) (deployed commit: 9fb922c0f518209cedbf4d883c4d9f38a98e604a).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/6d49e0f8fe1d238a6f9d8ca4aeb64e0d908bc0e8) (deployed commit: 9fb922c0f518209cedbf4d883c4d9f38a98e604a).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/4893edf422ca42df73a91963100590a6844a5238) (deployed commit: 9fb922c0f518209cedbf4d883c4d9f38a98e604a).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/350f31359ab0531fcd2919e5b64ca0df1a159ffd) (deployed commit: 9fb922c0f518209cedbf4d883c4d9f38a98e604a).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/6c66fc32fbaec949ba99bd58cd168fe9c100b883) (deployed commit: 9fb922c0f518209cedbf4d883c4d9f38a98e604a).

oliverguenther committed:
Update builds and shrinkwrap

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/27bd7a998561a36256d157b2c055f7dbf11039f6) (deployed commit: f3796c5ce8f2ea8e51bcf1a6ec2c714c243e5d19).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/9f33bc14587b26947569ddf5e0488e44177850db) (deployed commit: f3796c5ce8f2ea8e51bcf1a6ec2c714c243e5d19).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/2bc217cd622396c59f7fd36b2546853bdac32557) (deployed commit: f3796c5ce8f2ea8e51bcf1a6ec2c714c243e5d19).

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/commonmark-and-ckeditor

oliverguenther committed:
Add conversion for CF long text

oliverguenther committed:
Convert wiki links matcher from old text formatter into pattern matcher

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/8920bdbb3c7d932c67a0b3e4842120cf675d3a46) (deployed commit: dafb2e875f93c586cef0020ef592945b03a38d68).

oliverguenther committed:
Fix autolinks in rinku and add specs

oliverguenther committed:
Markdown requires continuation blockquote

In textile, jumping back from nested blockquote is possible with
```
> Quote
>>  subquote
> Quote
```

In CommonMark, subquotes continue
http://spec.commonmark.org/0.28/#example-213

so to continue in the parent quote, a second line is needed

```
> Quote
>>  Subquote
>
> Quote
```

oliverguenther committed:
Bump commonmarker (no longer requiring cmake)

oliverguenther committed:
Bump commonmarker version

[ci skip]

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/e05cd74dca035ccd433d3aa92b19b2dea2dfd2e3) (deployed commit: 6da76bd5b494d56419a4abb7e3d07560fcd059f7).

oliverguenther committed:
Bump pry-byebug to fix newline issue

oliverguenther committed:
Added markdown spec

oliverguenther committed:
Don't descend when rendering ### links

oliverguenther committed:
Make link regexp legible

oliverguenther committed:
Add group mention to markdown formatter

oliverguenther committed:
Skip PatternMatcher filter in code and pre blocks

oliverguenther committed:
Fix route generation for source urls and general escaping of quotes

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/7ac63ee5042f261284fc60ef655c308b75fc70bb) (deployed commit: 85a1079e25f02b3551d502038fedb60c859648e0).

oliverguenther committed:
Rest overridden edit fields

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/e1a286e2031b2d584d70520f31a87b5b8bb63a96) (deployed commit: 39c2dd4ac779ef20efe291caf352f5e81072e78b).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/1ef55f0a28e40c063dc24e669d3e63f9024d6d12) (deployed commit: c0b636affe1b210c6161034d9163e34359b8bdbb).

oliverguenther committed:
Add webpack-copy-plugin

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/a9b6bba8db193ace3ab71686c9fb00d058c36df1) (deployed commit: 27ce94d9cb07622e0f142b539fc5a2f134503332).

oliverguenther committed:
Fix repo tests due to changes repo_path

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/520d9dd521d304e623a9866eaffde52dfad9f71a) (deployed commit: 8fb1c1573d39763ea6186dc4def9d5dd99deafda).

oliverguenther committed:
Allow non-breaking-spaces in leading section of texts

This is due to nokogiri working on text fragments that may have nbsp in
them.

oliverguenther committed:
Upgrade at.js

oliverguenther committed:
Distinguish between textile and markdown spec

oliverguenther committed:
Convert user directive to edit in markdown mode

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/commonmark-and-ckeditor

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/57ea142ff4bc5638d8d3e2cdb8bf1d0a0d69d67a) (deployed commit: 8dfd802933f0ad5a400f4d0bc5838ea79ed02cc6).

oliverguenther committed:
Add vendored ckeditor builds for easy building an linking

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/3dc19fbde833e74813ec377b9b4464273683870e) (deployed commit: 6762b85d0bd94e492668b0f5e5645207600d079d).

oliverguenther committed:
Fix repo routing spec after path fix

oliverguenther committed:
Extract current text formatting helper into separate helper

oliverguenther committed:
Fix resource link parser regex to not include empty group member

oliverguenther committed:
Restore activity tab spec functionality

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/7b0c918e182b13cd186668c0b4e98fee93c8d466) (deployed commit: 23d60897a15bf06504081ae957b6bbe392b4a9c7).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/ca96b8b0d3e4c4de07f93c12aa836175ba739a80) (deployed commit: 13a13763af6e345795af13415af4f6366435e995).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/d5f47fc522b83f27afabfa563fc1fe3698f19396) (deployed commit: e93c49ecd7acdc1b06d41ba087d707050ba3853f).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/aadd9b6dac63163f2bae7d68b13f0ef0603f6638) (deployed commit: e93c49ecd7acdc1b06d41ba087d707050ba3853f).

oliverguenther committed:
Make link handlers overridable by plugins

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/8bc7c50d521f863ac491adb4327d5d4261ae055b) (deployed commit: c6c9ddacdafdc219d61ba5382e2ce4119f65b88c).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/69b05390b9ab08ae4be49463b76a942016356240) (deployed commit: f2939c75a1d965a7fd74c53e9e0a7266cc032665).

crohr commented:
A new preview environment is ready! [Click to open](https://pullpreview.com/opf/openproject/sessions/f53718439fcf4fdf55654467d8fa5e4906244e62) (deployed commit: f2939c75a1d965a7fd74c53e9e0a7266cc032665).

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/commonmark-and-ckeditor

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/commonmark-and-ckeditor

oliverguenther committed:
Create new beta build

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/commonmark-and-ckeditor

oliverguenther committed:
Fix tslint

oliverguenther committed:
Merge remote-tracking branch 'origin/dev' into feature/commonmark-and-ckeditor

oliverguenther committed:
Fix textile toc formatter error after merge
